go/vt/vtgate/vindexes/lookup.go

```diff
@@ -286,3 +353,44 @@ func (lu *LookupUnique) Delete(vcursor VCursor, rowsColValues [][]sqltypes.Value
 func (lu *LookupUnique) MarshalJSON() ([]byte, error) {
 	return json.Marshal(lu.lkp)
 }
+
+func numericUint64(input sqltypes.Value) ([]byte, error) {
+	// If attempting to cherry-pick this into a build <= 6.0 this needs to be
+	// rewritten as sqltypes.ToUint64
+	v, err := evalengine.ToUint64(input)
+	if err != nil {
+		return nil, fmt.Errorf("numericUint64: couldn't parse bytes: %v", err)
+	}
+	vBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(vBytes, v)
+	return vBytes, nil
+}
+
+func numericAsHexString(input sqltypes.Value) ([]byte, error) {
+	switch input.Type() {
+	case querypb.Type_INT8:
+	case querypb.Type_UINT8:
+	case querypb.Type_INT16:
+	case querypb.Type_UINT16:
+	case querypb.Type_INT24:
+	case querypb.Type_UINT24:
+	case querypb.Type_INT32:
+	case querypb.Type_UINT32:
+	case querypb.Type_INT64:
+	case querypb.Type_UINT64:
+	case querypb.Type_TEXT:
+	case querypb.Type_VARCHAR:
+	case querypb.Type_CHAR:
+	default:
+		return nil, fmt.Errorf("%v unsupported column type", input.Type())
+	}
+	str := string(input.ToBytes())
```

Comment:
i think you can do `input.ToString()` to take advantage of vitess' hacky string casting